LIMS2DB/classes.py

```diff
@@ -297,6 +316,20 @@ def build(self):
                 if 'Size (bp)' in agr_inp.udf_dict:
                     self.obj['projects'][project.luid]['samples'][sample.name]['library'][agr.luid]['size'] = round(agr_inp.udf_dict['Size (bp)'], 2)
 
+                #Grabbing indexes
+                # Get all artifacts for given sample
+                query = "select art.* from artifact art \
+                    inner join artifact_sample_map asm on asm.artifactid=art.artifactid \
+                    inner join sample sa on sa.processid=asm.processid \
+                    where sa.processid = {sapid};".format(sapid=sample.processid)
+                try:
+                    artifacts = self.session.query(Artifact).from_statement(text(query)).all()
+                    for art in artifacts:
+                            if art.reagentlabels is not None and len(art.reagentlabels) == 1:
```

Comment:
what happens if the length is larger than 1? well nothing ... but are there cases where art.reagnetlabels is larger than 1?